src/components/cli.ts

```diff
@@ -229,7 +233,24 @@ export class Cli<ConfigType extends Record<string, unknown>> {
         });
     }
 
-    private startWithConfig(config: ConfigType|null) {
+    private startWithConfig(config: ConfigType|null, configFilename: string) {
+        if (this.opts.onConfigChanged && this.opts.bridgeConfig) {
+            log.info("Will listen for SIGHUP");
+            process.on("SIGHUP",
+                () => {
+                log.info("Got SIGHUP, reloading config file");
+                try {
+                    const newConfig = this.loadConfig(configFilename);
+                    // onConfigChanged is checked above, but for Typescript's sake.
+                    if (this.opts.onConfigChanged) {
```

Comment:
Not just for TypeScript's sake. We're copying the `opts` object by reference.
Any code in this class or the bridge could have re-set the value of `opts.onConfigChanged` before `SIGHUP` was received.